src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Enums.cs

```diff
@@ -167,23 +168,122 @@ public enum FloatComparisonMode : byte
         TrueUnorderedSignaling = 31,
     }
 
+    [Flags]
     public enum StringComparisonMode : byte {
```

Comment:
Why do we need a regular `StringComparisonMode` as well? Aren't they all the methods either `Index` or `Mask`?